microprofile/connectors/kafka/src/main/java/io/helidon/microprofile/connectors/kafka/KafkaConnectorFactory.java

```diff
@@ -0,0 +1,130 @@
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.connectors.kafka;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.BeforeDestroyed;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import io.helidon.config.Config;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.spi.Connector;
+import org.eclipse.microprofile.reactive.messaging.spi.IncomingConnectorFactory;
+import org.eclipse.microprofile.reactive.messaging.spi.OutgoingConnectorFactory;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * Implementation of Connector as described in the MicroProfile Reactive Messaging Specification.
+ */
+@ApplicationScoped
+@Connector(KafkaConnectorFactory.CONNECTOR_NAME)
+class KafkaConnectorFactory implements IncomingConnectorFactory, OutgoingConnectorFactory {
+
+    /**
+     * Microprofile messaging Kafka connector name.
+     */
+    static final String CONNECTOR_NAME = "helidon-kafka";
+    private static final String POOL_SIZE = "kafka.connector.pool.size";
+    private static final Logger LOGGER = Logger.getLogger(KafkaConnectorFactory.class.getName());
+    private final ScheduledExecutorService scheduler;
+    private final Queue<BasicKafkaConsumer<Object, Object>> consumers = new LinkedList<>();
+    private final Queue<BasicKafkaProducer<Object, Object>> producers = new LinkedList<>();
+
+    @Inject
+    KafkaConnectorFactory(Config config) {
+        scheduler = Executors.newScheduledThreadPool(config.get(POOL_SIZE)
+                .asInt().asOptional().orElseGet(() -> 10));
+    }
+
+    /**
+     * Called when container is terminated.
+     *
+     * @param event termination event
+     */
+    public void terminate(@Observes @BeforeDestroyed(ApplicationScoped.class) Object event) {
```

Comment:
This method should be private or package local.

Reply:
Done